patches/minecraft/net/minecraft/client/gui/screen/CreateWorldScreen.java.patch

```diff
@@ -0,0 +1,12 @@
+--- a/net/minecraft/client/gui/screen/CreateWorldScreen.java
++++ b/net/minecraft/client/gui/screen/CreateWorldScreen.java
+@@ -384,7 +384,9 @@
+       if (path != null) {
+          File file1 = path.toFile();
+          ResourcePackList<ResourcePackInfo> resourcepacklist = new ResourcePackList<>(ResourcePackInfo::new, new ServerPackFinder(), new FolderPackFinder(file1, IPackNameDecorator.field_232625_a_));
++         net.minecraftforge.fml.packs.ResourcePackLoader.loadResourcePacks(resourcepacklist, net.minecraftforge.fml.server.ServerLifecycleHooks::buildPackFinder);
+          resourcepacklist.func_198983_a();
++         this.field_238933_b_ = DatapackCodec.getModdedDefault();
```

Comment:
This needs to 
1) Not be a random method shoved onto DatapackCodec
2) Make sure all uses of the DatapackCodec.field_234880_a_ uses our default mod pack list.
3) Not change this entry every time the button is pressed, but instead do it once in the initalizer and use the argument passed into the constructor like vanilla does.